src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -326,7 +328,7 @@ private string GetHeaderString(HeaderDescriptor descriptor, object info, object?
                     // the only collection that can be enumerated concurrently is HttpClient.DefaultRequestHeaders,
                     // and all values in it will be HeaderStoreItemInfo.
                     Debug.Assert(!_forceHeaderStoreItems);
-                    _headerStore[descriptor] = info = new HeaderStoreItemInfo() { RawValue = value };
+                    _headerStore[descriptor] = info = new HeaderStoreItemInfo() { RawValue = value, OriginalRawValue = value };
```

Comment:
It would be nice to assert here that value is a string. I know that's not in the existing code, but it seems like it should be added.